sashimi-webapp/src/logic/filemanager/folder.js

```diff
@@ -0,0 +1,174 @@
+import storage from '../../database/storage';
+import File from './file';
+
+/**
+* Folder Object
+*
+* @param {Number} folderID
+* @param {String} folderName
+* @param {String} folderPath
+*/
+
+function Folder(folderID, folderName, folderPath) {
+  this.id = folderID;
+  this.name = folderName;
+  this.path = folderPath;
+  this.parentFolder = null;
+  this.childFolderList = []; // list of folders in current folder
+  this.childFileList = []; // list of files in current folder
+  console.log('new Folder created');
+}
+
+/**
+ * Create a new folder in this folder instance
+ *
+ * @param {String} folderName
+ * @return {Folder}
+ */
+Folder.prototype.createFolder = function createFolder(folderName) {
+  console.log('folder.createFolder');
+  const newFolderPath = `${this.folderPath}/${folderName}`;
+  // const dbFolderObj = storage_api.createFolder(0, newFolderPath, this.id)
+  return storage.createFolder(0, newFolderPath, this.id)
+  .then((dbFolderObj) => {
+    const newFolder = new Folder(dbFolderObj.folder_ID, dbFolderObj.folderName, dbFolderObj.folderPath);
+    newFolder.parentFolder = this;
+    this.childFolderList.push(newFolder);
+    return newFolder;
+  })
+  .catch((error) => {
+    throw error;
+  });
+};
+
+/**
+ * Create a new file in this folder in this folder instance
+ *
+ * @param {String} fileName
+ * @return {File}
+ */
+Folder.prototype.createFile = function createFile(fileName) {
+  console.log('folder.createFile');
+  const newFilePath = `${this.folderPath}/${fileName}`;
+  return storage.createFile(0, newFilePath, this.id)
+  .then((dbFileObj) => {
+    const newFile = new File(dbFileObj.file_id, dbFileObj.file_name, dbFileObj.file_path, this);
+    this.childFileList.push(newFile);
+  })
+  .catch((error) => {
+    throw error;
+  });
+};
+
+/**
+ * Remove a folder from the database
+ *
+ * @param {}
+ * @return {}
+ */
+Folder.prototype.remove = function remove() {
+  console.log('folder.remove');
+  return storage.deleteFolder(this.id)
```

Comment:
By right user should not be able to delete the root folder... I will add in a check to ensure that the folder to remove is not the root folder.